pynautobot/models/dcim.py

```diff
@@ -213,7 +213,10 @@ def __str__(self):
         # hacky check to see if we're a circuit termination to
         # avoid another call to Nautobot because of a non-existent attr
         # in self.name
+        # Fetch details for the circuit if it doesn't have them
         if "circuit" in str(self.url):
+            if not self.has_details:
```

Comment:
Would it make sense here to move to a property?

Reply:
Not sure I understand.  Move `has_details` to a property?

Reply:
I think I get it or if not then it sparked another idea.  We can add `circuit` as a property to the `Termination` model I believe and that might be the better solution here.